registries/adapters/openshift_adapter_test.go

```diff
@@ -96,12 +97,23 @@ func TestRegistryName(t *testing.T) {
 	if err != nil {
 		t.Fatal("Error: ", err)
 	}
+	urlHttp, err := url.Parse(TestHttpURL)
+	if err != nil {
+		t.Fatal("Error: ", err)
+	}
+
 	ocpa := OpenShiftAdapter{
 		Config: Configuration{
 			URL: url,
 		},
 	}
+	ocpaHttp := OpenShiftAdapter{
+		Config: Configuration{
+			URL: urlHttp,
+		},
+	}
 	ft.Equal(t, ocpa.RegistryName(), RegistryName, "the registry name returned did not match expected value")
+	ft.Equal(t, ocpaHttp.RegistryName(), RegistryName, "the registry name returned did not match expected value")
 }
```

Comment:
Consider making this a table driven testcase.

```golang
func TestRegistryName(t *testing.T) {

    testcases := []struct {
        urlstr string
        result string
    }{
        {"http://foo.redhat.com", "foo.redhat.com"},
        {"https://registry.reg-aws.openshift.com:443", "registry.reg-aws.openshift.com:443"},
        {"ftp://registry.reg-aws.openshift.com", "registry.reg-aws.openshift.com"},
        {"file:///home/user/awesome", "/home/user/awesome"},
    }

    for _, testcase := range testcases {
        url, err := url.Parse(testcase)
        if err != nil {
            t.Fatal("Error: ", err)
        }

        ocpa := OpenShiftAdapter{
            Config: Configuration{
                URL: url,
            },
        }
        ft.Equal(t, ocpa.RegistryName(), result, "the registry name returned did not match expected value")
    }
}
```

Reply:
Thanks that does look much better. Appreciate the feedback.